greenplumpython/db.py

```diff
@@ -11,7 +11,9 @@ def __init__(self, **params) -> None:
             cursor_factory=psycopg2.extras.RealDictCursor,
         )
 
-    def execute(self, query: str, args: Iterable = [], has_results: bool = True) -> Iterable:
+    def execute(
+        self, query: str, args: Optional[List[str]] = None, has_results: bool = True
```

Comment:
`args` doesn't have to be `Optional`, `List` should be enough

```
    def execute(self, query: str, args: List = [], has_results: bool = True) -> Optional[Iterable]:
        with self._conn.cursor() as cursor:
            cursor.execute(query, args)
            return cursor.fetchall() if has_results else None
```
Would the above be enough to fix the lint issue?

Why does `args` need to be `List[str]`? Actually, I don't think we need the `args` param, do we? @xuebinsu

Reply:
This depends on whether the database driver exposes the function to convert a Python constant, e.g. a byte array, to a Postgres constant.
- If yes, then we do not need this argument, anything can be converted to a string.
- If no, then we need to use the query string as only the template and pass the arguments separately.

Reply:
```suggestion
            def execute(self, query: str, args: List = [], has_results: bool = True) -> Optional[Iterable]:
```